common-codegen/src/main/java/io/vrap/rmf/codegen/common/generator/model/immutable/ImmutableModelGenerator.java

```diff
@@ -0,0 +1,227 @@
+package io.vrap.rmf.codegen.common.generator.model.immutable;
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.squareup.javapoet.*;
+import io.reactivex.Flowable;
+import io.reactivex.Single;
+import io.vrap.rmf.codegen.common.generator.core.CodeGenerator;
+import io.vrap.rmf.codegen.common.generator.core.GenerationResult;
+import io.vrap.rmf.codegen.common.generator.core.GeneratorConfig;
+import io.vrap.rmf.codegen.common.generator.doc.JavaDocProcessor;
+import io.vrap.rmf.codegen.common.generator.util.CtAnnotationProcessor;
+import io.vrap.rmf.raml.model.modules.Api;
+import io.vrap.rmf.raml.model.types.*;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.emf.ecore.util.EObjectContainmentEList;
+import org.immutables.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.lang.model.element.Modifier;
+import javax.tools.JavaFileObject;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.vrap.rmf.codegen.common.generator.util.CodeGeneratorUtil.*;
+
+public class ImmutableModelGenerator extends CodeGenerator {
+
+    protected final Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+    public ImmutableModelGenerator(GeneratorConfig generatorConfig, Api api) {
+        super(generatorConfig, api);
+    }
+
+    private static String getMethodName(final Property property) {
+        final String javaPropertyName = Arrays.stream(property.getName().split("_"))
+                .map(StringUtils::capitalize)
+                .collect(Collectors.joining());
+        if (property.getType() instanceof BooleanType) {
+            return "is" + javaPropertyName;
+        }
+        return "get" + javaPropertyName;
+    }
+
+    public List<Path> getPaths(List<JavaFileObject> javaFiles, Path outputFolder) {
+        return javaFiles.stream().map(javaFile -> Paths.get(outputFolder.toString(), javaFile.getName())).collect(Collectors.toList());
+    }
+
+    protected Stream<AnnotationSpec> getDiscriminatorAnnotations(final ObjectType object) {
```

Comment:
Here we should rename the parameter to `objectType` too.

Reply:
nice catch 👍